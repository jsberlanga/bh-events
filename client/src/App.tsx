import * as React from 'react';
import { EventList } from './components/EventList';
import styled from 'styled-components';

const AppContainer = styled.div`
  margin: 0 auto;
  width: min(50rem, 100vw);
  padding: 1rem;
`;

function App() {
  return (
    <AppContainer>
      <EventList />
    </AppContainer>
  );
}

export default App;
