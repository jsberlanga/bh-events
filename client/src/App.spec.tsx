import { render } from '@testing-library/react';
import App from './App';

describe('App', () => {
  it('renders with styles', () => {
    const { container } = render(<App />);

    expect(container).toMatchSnapshot();
  });
});
