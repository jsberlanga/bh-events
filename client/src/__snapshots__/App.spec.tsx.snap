// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders with styles 1`] = `
.c4 {
  list-style: none;
  padding: 0;
  display: grid;
  grid-template-columns: 1fr;
  grid-gap: 1rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 15rem;
  margin-bottom: 1rem;
  padding: 14px;
}

.c2 {
  font-size: 0.8rem;
  margin-bottom: 0.7rem;
}

.c3 {
  box-sizing: border-box;
  width: 100%;
  padding: 6px;
  font-size: 1rem;
}

.c0 {
  margin: 0 auto;
  width: min(50rem,100vw);
  padding: 1rem;
}

.c0 header {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 > * {
  margin-bottom: 2rem;
}

@media screen and (min-width:768px) {
  .c4 {
    grid-template-columns: repeat(3,33%);
  }
}

<div>
  <div
    class="c0"
  >
    <header>
      <svg
        height="16px"
        viewBox="0 0 114.06 15.12"
        width="119px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="#f76ed6"
          height="3.33"
          transform="translate(0.38 2.96) rotate(-90)"
          width="2.58"
          x="0.38"
          y="-0.38"
        />
        <rect
          fill="#f76ed6"
          height="2.58"
          transform="translate(-5.36 21.63) rotate(-90)"
          width="2.58"
          x="6.84"
          y="12.2"
        />
        <rect
          fill="#f76ed6"
          height="2.58"
          transform="translate(63.57 66.15) rotate(-90)"
          width="2.58"
          x="63.57"
        />
        <polygon
          fill="#5331d5"
          points="14.09 0 11.3 0 6.02 0 6.02 2.58 11.3 2.58 14.09 2.58 20.93 2.58 20.93 0 14.09 0"
        />
        <rect
          fill="#5331d5"
          height="4.29"
          transform="translate(-11.35 15.64) rotate(-90)"
          width="2.58"
          x="0.86"
          y="11.35"
        />
        <rect
          fill="#55d7ff"
          height="10.83"
          transform="translate(1.29 16.07) rotate(-90)"
          width="2.58"
          x="7.39"
          y="1.98"
        />
        <path
          d="M86.66,4.41H86.2a9.44,9.44,0,0,0-2.26.36V.69H81.28V14.79h2.66v-7a3.08,3.08,0,0,1,2.26-.84,1.25,1.25,0,0,1,1.07.43,2.44,2.44,0,0,1,.32,1.43v6h2.69V8.68C90.27,6,89.06,4.5,86.66,4.41Z"
          fill="#131740"
        />
        <path
          d="M110.45,4.41H110a9.44,9.44,0,0,0-2.26.36V.69h-2.66V14.79h2.66v-.36A5.74,5.74,0,0,0,110,15h.46c2.4-.09,3.62-1.6,3.62-4.27V8.68C114.06,6,112.85,4.5,110.45,4.41Zm.93,6.24a2.44,2.44,0,0,1-.32,1.43,1.25,1.25,0,0,1-1.07.43,3.08,3.08,0,0,1-2.26-.84V7.74A3.08,3.08,0,0,1,110,6.89a1.25,1.25,0,0,1,1.07.43,2.44,2.44,0,0,1,.32,1.43Z"
          fill="#131740"
        />
        <path
          d="M55,4.41c-2.4.09-3.62,1.6-3.62,4.27v2.05c0,2.67,1.21,4.18,3.62,4.27h.46a7.19,7.19,0,0,0,2.26-.46v.4h2.66V6.2a7.46,7.46,0,0,0-4.91-1.78Zm2.72,7.26a3.08,3.08,0,0,1-2.26.84,1.25,1.25,0,0,1-1.07-.43,2.45,2.45,0,0,1-.32-1.43V8.76a2.45,2.45,0,0,1,.32-1.43,1.25,1.25,0,0,1,1.07-.43,3.08,3.08,0,0,1,2.26.84Z"
          fill="#131740"
        />
        <rect
          fill="#131740"
          height="9.97"
          width="2.64"
          x="63.54"
          y="4.82"
        />
        <path
          d="M40.1,7.19A4.32,4.32,0,0,0,36.85,0H31V14.79h6.22a4.32,4.32,0,0,0,2.85-7.6ZM36.85,2.58a1.76,1.76,0,1,1,0,3.52H33.62V2.58Zm.4,9.62H33.62V8.68h3.64a1.76,1.76,0,1,1,0,3.52Z"
          fill="#131740"
        />
        <path
          d="M74.61,4.41h-.46A7.46,7.46,0,0,0,69.24,6.2v8.58h2.66v-7a3.08,3.08,0,0,1,2.26-.84,1.25,1.25,0,0,1,1.07.43,2.44,2.44,0,0,1,.32,1.43v6h2.69V8.68C78.23,6,77,4.5,74.61,4.41Z"
          fill="#131740"
        />
        <path
          d="M99.14,11.79a3.08,3.08,0,0,1-2.26.84,1.25,1.25,0,0,1-1.07-.43,2.45,2.45,0,0,1-.32-1.43v-6H92.8v6.11c0,2.67,1.21,4.18,3.62,4.27h.46a7.46,7.46,0,0,0,4.91-1.78V4.74H99.14Z"
          fill="#131740"
        />
        <path
          d="M48.78,4.69a7.46,7.46,0,0,0-4.91,1.78v8.32h2.66V8a3.16,3.16,0,0,1,2.27-.89,6.18,6.18,0,0,1,1.14.09V4.69H48.78Z"
          fill="#131740"
        />
        <rect
          fill="#f76ed6"
          height="4.26"
          transform="translate(11.41 26.2) rotate(-90)"
          width="2.58"
          x="17.51"
          y="5.26"
        />
        <rect
          fill="#55d7ff"
          height="8.89"
          transform="translate(2.99 29.98) rotate(-90)"
          width="2.58"
          x="15.2"
          y="9.05"
        />
      </svg>
      <a
        href="https://github.com/jsberlanga/bh-events"
      >
        Code in Github
      </a>
    </header>
    <h1>
      EVENTS CHALLENGE
    </h1>
    <hr />
    <h2>
      Create a new event
    </h2>
    <form
      class="c1 withBorder"
    >
      <label
        class="c2"
        for="name"
      >
        Name
        <input
          class="c3 withBorder"
          id="name"
          required=""
          type="text"
          value=""
        />
      </label>
      <label
        class="c2"
        for="lastname"
      >
        Lastname
        <input
          class="c3 withBorder"
          id="lastname"
          required=""
          type="text"
          value=""
        />
      </label>
      <label
        class="c2"
        for="email"
      >
        Email
        <input
          class="c3 withBorder"
          id="email"
          required=""
          type="email"
          value=""
        />
      </label>
      <label
        class="c2"
        for="date"
      >
        Date
        <input
          class="c3 withBorder"
          id="date"
          required=""
          type="date"
          value="2021-01-21"
        />
      </label>
      <input
        class="c3 btn btn--submit"
        type="submit"
        value="create"
      />
    </form>
    <hr />
    <h2>
      List with all recorded events
    </h2>
    <ul
      class="c4"
    >
      There are no events yet.
    </ul>
  </div>
</div>
`;
